engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -111,12 +111,7 @@ public static AnalyzerContext createContext(
 
         // First pass to initialize all columns and to compile formulas in one batch.
         final QueryCompilerRequestProcessor.BatchProcessor compilationProcessor = QueryCompilerRequestProcessor.batch();
-        for (Map.Entry<String, ColumnSource<?>> entry : columnSources.entrySet()) {
-            final String name = entry.getKey();
-            final ColumnSource<?> cs = entry.getValue();
-            final ColumnDefinition<?> cd = ColumnDefinition.fromGenericType(name, cs.getType(), cs.getComponentType());
-            columnDefinitions.put(name, cd);
-        }
+        parentTable.getDefinition().getColumns().forEach(cd -> columnDefinitions.put(cd.getName(), cd));
```

Comment:
I would definitely run nightlies. I would also like you to test out what happens if you run `select()`, `view()`, `update()`, and `updateView()` preserving a partitioning column: does the output reflect that the column is partitioning, and is that a change?